docs/reference/function-configuration/code-entry-types.md

```diff
@@ -0,0 +1,245 @@
+
+The downloaded function code can optionally contain a **function.yaml** file with function configuration for enriching the original configuration (in the configuration file that sets the code-entry type) according to the following merge strategy:
+
+- Field values that are set only in the downloaded configuration are added to the original configuration.
+- List and map field values &mdash; such as `meta.labels` and `spec.env` &mdash;are merged by adding any values that are set only in the downloaded configuration to the values that are set in the original configuration.
+- In case of a conflict &mdash; i.e., if the original and downloaded configurations set different values for the same element &mdash; the original configuration takes precedence and the value in the downloaded configuration is ignored.
+
+<a id="code-entry-type-github"></a>
+### GitHub code-entry type (`github`)
+
+Set the [`spec.build.codeEntryType`](/docs/reference/function-configuration.md#spec.build.codeEntryType) function-configuration field to `github` (dashboard: **Code entry type** = `GitHub`) to download the function code from a GitHub repository. The following configuration fields provide additional information for performing the download:
+
+- `spec.build` &mdash;
+  - `path` (dashboard: **URL**) (Required) &mdash; the URL of the GitHub repository that contains the function code.
+  - `codeEntryAttributes` &mdash;
+    -  `branch` (dashboard: **Branch**) (Required) &mdash; the GitHub repository branch from which to download the function code.
+    - `headers.Authorization` (dashboard: **Token**) (Optional) &mdash; a GitHub access token for download authentication.
+    - `workDir` (dashboard: **Work directory**) (Optional) &mdash; the relative path to the function-code directory within the configured repository branch.
+
+<a id="code-entry-type-github-example"></a>
+#### Example
+
+```yaml
+spec:
+  description: my Go function
+  handler: main:Handler
+  runtime: golang
+  build:
+    codeEntryType: "github"
+    path: "https://github.com/my-organization/my-repository"
+    codeEntryAttributes:
+      branch: "my-branch"
+      headers:
+        Authorization: "my-Github-access-token"
+      workDir: "/go/myfunc"
+```
+
+<a id="code-entry-type-archive"></a>
+### Archive-file code-entry type (`archive`)
+
+Set the [`spec.build.codeEntryType`](/docs/reference/function-configuration.md#spec.build.codeEntryType) function-configuration field to `archive` (dashboard: **Code entry type** = `Archive`) to download a ZIP archive file of the function code either from an [Iguazio Data Science Platform](https://www.iguazio.com) ("platform") data container or from any URL that doesn't require download authentication. The following configuration fields provide additional information for performing the download:
```

Comment:
>  ~ZIP~ archive - I think stating "zip" in the config might confuse as per the supported formats and archive extensions.
I suggest to add a list of supported formats (@sahare92 should be able to provide them based on the package we use in the code here)

Reply:
I was incorrectly under the impression that the archive-file options were restricted to ZIP ... . Sahar has now sent me the list of supported archive formats (`.jar .rar .zip .tar .tar.bz2 .tar.gz .tar.lz4 .tar.sz .tar.xz`) and I edited the doc to remove the specific ZIP references (except in the examples); add a supported formats note in the "External function-code entry types" section (#archive-file-formats); and link to this note from the descriptions of the `archive` and `s3` types. See commit b9df093.

Reply:
Awesome, thank you!